### PR TITLE
Create action to destroy store - 136346095

### PR DESCRIPTION
**Pull request by rodrigoulisses**
Only admin can destroy a store.
Deleting a store, destroy all maps and users.

### CODE REVIEW

File: app/controllers/stores_controller.rb [hunk 1]
```diff
@@ -11,12 +11,19 @@ def create
...
     if @store.save
-      redirect_to edit_store_map_path(store_id: @store.id, id: @store.maps.last.id)
+      redirect_to edit_store_map_path(store_id: @store.id, id: @store.maps.last.id), notice: I18n.t('stores.notice.create')
     else
       render :new
     end
   end
 
+  def destroy
+    store = Store.find(params[:id])
```
Comment (cyruscavalcante) [inline app/controllers/stores_controller.rb]: Store.find(params[:id]).destroy

File: spec/controllers/stores_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe StoresController, type: :controller do
+  let!(:store) { create(:store_with_maps, name: 'Leroy') }
```
Comment (marclerodrigues) [inline spec/controllers/stores_controller_spec.rb]: You don't need the bang here a believe.

File: spec/controllers/stores_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    it 'delete store' do
+      expect do
+        delete :destroy, params: { id: store.id }
+      end.to change{Store.count}.by(-1)
+    end
+
+    it 'delete store maps' do
+      expect do
+        delete :destroy, params: { id: store.id }
+      end.to change{Map.count}.by(-1)
```
Comment (marclerodrigues) [inline spec/controllers/stores_controller_spec.rb]: The same here.

File: spec/controllers/stores_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    end
+
+    it 'delete store users' do
+      expect do
+        delete :destroy, params: { id: store.id }
+      end.to change{User.count}.by(-1)
```
Comment (marclerodrigues) [inline spec/controllers/stores_controller_spec.rb]: And here.

File: spec/controllers/stores_controller_spec.rb (hunk 2 above)
Comment (dmitryrck) [inline spec/controllers/stores_controller_spec.rb]: Neither in `user` :P

File: spec/business/destroy_store_spec.rb [hunk 5]
```diff
@@ -0,0 +1,57 @@
+require "rails_helper"
+
+RSpec.describe DestroyStore do
+  let!(:user) { create(:user) }
```
Comment (marclerodrigues) [inline spec/business/destroy_store_spec.rb]: Try without the bang in both of these let's.

**Pull request merged**